page-tracking/src/collections/page_box.rs

```diff
@@ -49,52 +51,67 @@ use crate::PageTracker;
 pub struct PageBox<T> {
     ptr: NonNull<T>,
     page_size: PageSize,
+    page_count: u64,
     page_tracker: PageTracker,
 }
 
 impl<T> PageBox<T> {
-    /// Creates a `PageBox` that wraps the given data using `page` to store it, returning it to its
+    /// Creates a `PageBox` that wraps the given data using `pages` to store it, returning it to its
     /// previous owner on `drop()` using `page_tracker`.
-    pub fn new_with(data: T, page: Page<InternalClean>, page_tracker: PageTracker) -> Self {
-        let ptr = page.addr().bits() as *mut T;
+    pub fn new_with(
+        data: T,
+        pages: SequentialPages<InternalClean>,
+        page_tracker: PageTracker,
+    ) -> Self {
+        let ptr = pages.base().bits() as *mut T;
         assert!(!ptr.is_null()); // Explicitly ban pages at zero address.
         unsafe {
             // Safe as the memory is totally owned and T fits in this page.
-            assert!(core::mem::size_of::<T>() <= page.size() as usize);
+            assert!(core::mem::size_of::<T>() <= pages.length_bytes() as usize);
             core::ptr::write(ptr, data);
         }
         Self {
             ptr: NonNull::new(ptr).unwrap(),
-            page_size: page.size(),
+            page_size: pages.page_size(),
+            page_count: pages.len(),
             page_tracker,
         }
     }
 
-    /// Consumes the `PageBox` and returns the page that was used to hold the data.
-    pub fn to_page(mut self) -> Page<InternalDirty> {
-        let page = unsafe {
+    /// Creates a `PageBox` with a single `page` to store it.
+    pub fn new_with_single_page(
+        data: T,
+        page: Page<InternalClean>,
+        page_tracker: PageTracker,
+    ) -> Self {
+        PageBox::new_with(data, SequentialPages::from(page), page_tracker)
```

Comment:
optional: given the `From` impl for `Page` to `SequentialPages` I'm not sure we need the separate `new_with_single_page()`.

Reply:
My reasoning was that "PageBox" as a type does strongly suggest that there's a single page behind a structure. And since it was the only use right now in the code I kind of wanted to keep the interface for it. No strong opinion, will remove it.